Apps/Playground/iOS/ViewController.swift

```diff
@@ -1,6 +1,71 @@
 import UIKit
 import MetalKit
 
+import UIKit
+import UIKit.UIGestureRecognizerSubclass
+
+/**
+  * A very simple gesture recognizer.  All that it does is to emulate the functionality found in other platforms
+  */
+class UIBabylonGestureRecognizer: UIGestureRecognizer {
+    // Callback for touch down events
+    public var onTouchDown: (Int32, Int32, Int32)->Void = {(pointerId: Int32, inX: Int32, inY: Int32) -> Void in
+        // Do nothing
+    }
+    // Callback for touch movement events
+    public var onTouchMove: (Int32, Int32, Int32)->Void = {(pointerId: Int32, inX: Int32, inY: Int32) -> Void in
+        // Do nothing
+    }
+    // Callback for touch up events
+    public var onTouchUp: (Int32, Int32, Int32)->Void = {(pointerId: Int32, inX: Int32, inY: Int32) -> Void in
+        // Do nothing
+    }
+    // Table to track hashes of active touches
+    private var _activeTouchIds: Array<Int> = [-1, -1, -1, -1, -1, -1, -1, -1, -1, -1]
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent) {
+        super.touchesBegan(touches, with: event)
+        
+        for touch in touches {
+            let deviceSlot = _activeTouchIds.firstIndex(of: -1)
+            
+            if (deviceSlot != nil) {
+                _activeTouchIds[deviceSlot!] = touch.hash
```

Comment:
What happens if there are more than 10 ids? Should we check this? Does swift arrays automatically check?

Reply:
If all of the entries in the `_activeTouchIds` are not -1 (all taken) the `firstIndex` should return `nil` so it will work like web and ignore additional touches